### PR TITLE
CI: Add container registry

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

CI: add the registry container, so we can run tests that need a non-authorized container

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed

- [X] **DONE**

## Test coverage
- No tests
- [X] **DONE**

## Links

Issue(s):https://github.com/SUSE/spacewalk/issues/20677

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
The registry container is just the official registry from docker hub. However, I want to have this in our github container registry, so that, when we run our tests, we do not rely on docker hub. Note that docker hub has a limit of pulls.

File: testsuite/dockerfiles/container-registry/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM registry:2
```
Comment (nodeg) [inline testsuite/dockerfiles/container-registry/Dockerfile:1]: Just curious, what is exactly behind the number 2? How does Docker know what image is behind this number?

Comment (jordimassaguerpla) [reply testsuite/dockerfiles/container-registry/Dockerfile:1]: It is registry version 2

**Comment by jordimassaguerpla** [comment]
Superseded by https://github.com/uyuni-project/uyuni/pull/8929

**Pull request closed without merging**